### ISSUE TITLE
404 page HTTP status code 200?

### ISSUE THREAD

**Issue by vdraceil** [gridsome/gridsome#831]
When the user gets redirected to a 404 page for obvious reasons, the page renders with a HTTP 200 status code.

I understand that the route matches with a valid page (`src/pages/404.vue`) and hence technically it is a successful page hit and hence a HTTP 200, but actually that is still a real 404.

Shouldn't the 404 page be treated as a special type of page instead?

Or is there already a page level config / prop setting which will help override the page status code?
like Nuxt which has this `error.statusCode` prop setting in their error layouts.

**Comment by jevy** [comment]
Did you find a solution for this at all @vdraceil?

**Comment by vdraceil** [comment]
> Did you find a solution for this at all @vdraceil?

No, I just left my 404 page as such (HTTP 200). I tried but couldn't find any workarounds.
I guess, this has to be a core change - or at least seems so.

**Comment by noxify** [comment]
Hi,

i think this can't be handled via gridsome itself.
We only have html files and it's not possible to set the correct status code in a html file.

https://stackoverflow.com/questions/8890017/how-to-return-specific-http-status-code-in-a-plain-html-page

Based on what i have seen, you can configure it via a `.htaccess` file.

**Comment by vdraceil** [comment]
.. but Nuxt is able to allow us an override, right? https://nuxtjs.org/guide/async-data#handling-errors
So, it should be possible from within the framework, I suppose.

**Comment by noxify** [comment]
Yes Nuxt allows you this but gridsome isn’t Nuxt ;) 

I think the comment from @tomdev here https://github.com/gridsome/gridsome/issues/193#issuecomment-464012508 should explain the differences between Gridsome and Nuxt.

**Comment by styfle** [comment]
This works with [Next.js](https://nextjs.org/docs/advanced-features/custom-error-page) too and many other frameworks.

The problem I see is that `gridsome develop` needs to set the status code to 404 whenever it serves the `src/pages/404.vue` page due to a missed route.

**Comment by belvederef** [comment]
I have tried to find a workaround to this to no avail. It is a serious problem as we transitioned an old website to gridsome, and in the process we decided to get rid of a lot of pages. Now search engines still show our deleted pages in the search results, because every time they reindex they are welcomed with HTTP 200 codes and because of that everything looks fine on their side.

**Comment by seangwright** [comment]
Nuxt serves content dynamically (when not doing pre-rendering) which means there is a web application running on a server that can set HTTP headers of the content being sent back to the browser, and it can set the status code for 'not found' pages to 404.

Gridsome generates static assets. Static assets can't set HTTP status codes - that requires a server/application to send the status code in the HTTP response.

The host that serves the Gridsome generated assets would have to be configured to serve the 404.html, that Gridsome creates, with a 404 status code in the HTTP response.
